api/layer/object_cache.go

```diff
@@ -74,6 +86,31 @@ func (l *listObjectsCache) Put(key string, objects []*ObjectInfo) {
 	})
 }
 
-func createKey(accessKey string, cid *cid.ID) string {
-	return accessKey + cid.String()
+func createKey(ctx context.Context, cid *cid.ID) (string, error) {
+	box, err := GetBoxData(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return box.Gate.AccessKey + cid.String(), nil
+}
+
+func trimAfterObjectName(startAfter string, objects []*ObjectInfo) []*ObjectInfo {
+	if len(startAfter) != 0 && len(objects) != 0 && objects[0].Name <= startAfter {
+		for i := range objects {
+			if objects[i].Name > startAfter {
+				return objects[i:]
+			}
+		}
+	}
+	return objects
```

Comment:
Why we return full list rather than empty if all key before `startAfter`?
For example if we have list `[]string{"a", "b"}` and pass `startAfter=c` I expect the result will be `[]string{}`

Reply:
You're right, writing unit tests...

Reply:
I think we can simplify this function to:
```
func trimAfterObjectName(startAfter string, objects []*ObjectInfo) []*ObjectInfo {
	for i := range objects {
		if objects[i].Name > startAfter {
			return objects[i:]
		}
	}
	return []*ObjectInfo{}
}
```

Reply:
Yep, I've made this changes but have not pushed yet. I will re-request review when I finish.